### PR TITLE
[Blogging prompts v1] Implement introduction dialog analytics events

### LINKED ISSUE #16521: [Blogging prompts v1] Implement analytics events
Implement analytics events in the following scenarios:

- [x]  Introduction modal #16525 
- [ ] Introduction notification
- [x] "My Site" prompt card #16536 
- [x] Blogging reminder modal #16634 16634
- [x] Blogging reminder notification #16579 
- [x] "Add new" bottom sheet #16657
- [x] Editor #16657

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #16521

To test:
- Enable blogging prompts feature flag
- Open app logged in and with a site selected
- Click on the blogging prompts onboarding notification OR in the "Learn more" button
- Blogging prompts onboarding dialog should be seen
  - `"blogging_prompts_introduction_modal_viewed"` event should be triggered when the dialog is shown
  - `"blogging_prompts_introduction_modal_dismissed"` event should be triggered when user taps on "x" button or back button
  - `"blogging_prompts_introduction_modal_try_it_now_tapped"` event should be triggered when user taps on "Try it now" button
  - `"blogging_prompts_introduction_modal_remind_me_tapped"` event should be triggered when user taps on "Remind me" button
  - `"blogging_prompts_introduction_modal_got_it_tapped"` event should be triggered when user taps on "Got it" button

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated `BloggingPromptsOnboardingViewModelTest.kt` and implemented `FeatureIntroductionViewModelTest.kt`.

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingViewModel.kt [hunk 1]
```diff
@@ -32,6 +33,7 @@ class BloggingPromptsOnboardingViewModel @Inject constructor(
     @Suppress("MaxLineLength")
     /* ktlint-disable max-line-length */
     fun start(type: DialogType) {
+        analyticsTracker.trackScreenShown()
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingViewModel.kt]: If I understand this correctly, this tracker will also be triggered on configuration change? We probably want to fire it only once :)

**Review by khaykov — COMMENTED** [review]
Great idea to use FeatureIntroductionViewModel for events in underlying dialog! I left one comment in the code, but the rest looks good to me 👍

**Comment by RenanLukas** [comment]
Thanks for the review @khaykov 
I've applied the suggestion you made 👍

**Review by khaykov — APPROVED** [review]
Thanks for the changes, looks good 👍

**Comment by AliSoftware** [comment]
Merging this myself as I'm about to do the `19.9` code freeze so might as well include this now that it's approved 🙂

**Pull request merged**